pkg/cmd/admin/diagnostics/network_pod.go

```diff
@@ -0,0 +1,175 @@
+	flag "github.com/spf13/pflag"
+
+	kcmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+	kutilerrors "k8s.io/kubernetes/pkg/util/errors"
+	"k8s.io/kubernetes/pkg/util/sets"
+
+	"github.com/openshift/origin/pkg/cmd/admin/diagnostics/options"
+	"github.com/openshift/origin/pkg/cmd/admin/diagnostics/util"
+	osclientcmd "github.com/openshift/origin/pkg/cmd/util/clientcmd"
+	"github.com/openshift/origin/pkg/diagnostics/log"
+	networkdiag "github.com/openshift/origin/pkg/diagnostics/networkpod"
+	"github.com/openshift/origin/pkg/diagnostics/types"
+)
+
+// NetworkPodDiagnosticsOptions holds values received from environment variables
+// for the command to operate.
+type NetworkPodDiagnosticsOptions struct {
+	// list of diagnostic names to limit what is run
+	RequestedDiagnostics []string
+	// LogOptions determine globally what the user wants to see and how.
+	LogOptions *log.LoggerOptions
+	// The Logger is built with the options and should be used for all diagnostic output.
+	Logger *log.Logger
+}
+
+const longNetworkPodDiagDescription = `
+This utility is intended to run network diagnostics inside a privileged container and
+log the results so that the calling diagnostic can report them.
+`
+
+var (
+	// availableNetworkPodDiagnostics contains the names of network diagnostics that can be executed
+	// during a single run of diagnostics. Add more diagnostics to the list as they are defined.
+	availableNetworkPodDiagnostics = sets.NewString(networkdiag.CheckNodeNetworkName, networkdiag.CheckPodNetworkName, networkdiag.CheckExternalNetworkName, networkdiag.CheckServiceNetworkName, networkdiag.CollectNetworkInfoName)
+)
+
+// NewCommandNetworkPodDiagnostics is the command for running network diagnostics.
+func NewCommandNetworkPodDiagnostics(name string, out io.Writer) *cobra.Command {
+	o := &NetworkPodDiagnosticsOptions{
+		RequestedDiagnostics: []string{},
+		LogOptions:           &log.LoggerOptions{Out: out},
+	}
+
+	cmd := &cobra.Command{
+		Use:   name,
+		Short: "Within a privileged pod, run network diagnostics",
+		Long:  fmt.Sprintf(longNetworkPodDiagDescription),
+		Run: func(c *cobra.Command, args []string) {
+			kcmdutil.CheckErr(o.Complete(args))
+
+			failed, err, warnCount, errorCount := o.BuildAndRunDiagnostics()
+			o.Logger.Summary(warnCount, errorCount)
+
+			kcmdutil.CheckErr(err)
+			if failed {
+				os.Exit(255)
+			}
+
+		},
+	}
+	cmd.SetOutput(out) // for output re: usage / help
+
+	options.BindLoggerOptionFlags(cmd.Flags(), o.LogOptions, options.RecommendedLoggerOptionFlags())
+
+	return cmd
+}
+
+// Complete fills in NetworkPodDiagnosticsOptions needed if the command is actually invoked.
+func (o *NetworkPodDiagnosticsOptions) Complete(args []string) (err error) {
+	o.Logger, err = o.LogOptions.NewLogger()
+	if err != nil {
+		return err
+	}
+
+	o.RequestedDiagnostics = append(o.RequestedDiagnostics, args...)
+	if len(o.RequestedDiagnostics) == 0 {
+		o.RequestedDiagnostics = availableNetworkPodDiagnostics.List()
```

Comment:
Yes, this was intentionally done this way. Collecting network info only in case of errors needs another round of launching network diagnostic pod (with different sub-command to openshift infra)  and waiting for it to finish the collection. This step is taking more time than collecting network info as part of other network diagnostic checks.